chassis/core/src/main/java/com/griddynamics/jagger/master/HibernateSessionIdProvider.java

```diff
@@ -23,20 +23,20 @@
 import java.util.List;
 
 import org.springframework.orm.hibernate3.HibernateTemplate;
+import com.griddynamics.jagger.exception.TechnicalException;
 import org.springframework.orm.hibernate3.support.HibernateDaoSupport;
 
-import com.griddynamics.jagger.exception.TechnicalException;
+import java.util.List;
 
 public class HibernateSessionIdProvider extends HibernateDaoSupport implements SessionIdProvider {
-	private final Object lock = new Object();
 	private String sessionId;
     private String sessionName;
     private String sessionComment;
 
 	@Override
 	public String getSessionId() {
 		if (sessionId == null) {
-			synchronized (lock) {
+			synchronized (this) {
```

Comment:
The way I see it - I fixed data inconsistency here. 
After you release a lock on private object in this particular piece of code there are no guarantee next thread which enter this critical section will see updated reference to sessionId, i.e. it can still read it from tread local cache and see it as null and initialise it again. But when a thread acquires a lock on this object it forced to reread its state from main memory (so, it acts like if sessionId was volatile)